test/functional/kernel/nested-loop/test-kernel-nested-loop.cpp.in

```diff
@@ -0,0 +1,154 @@
+
+#include "test-kernel-nested-loop-@NESTED_LOOP_TYPE@.hpp"
+
+// ----------------------------------------------------------------------------
+// Sequential
+// ----------------------------------------------------------------------------
+using SequentialKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::seq_exec  >,
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::loop_exec >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::seq_exec  >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::loop_exec >,
+
+    // Depth 3 Exec Pols
+    NestedLoopData<DEPTH_3, RAJA::seq_exec,  RAJA::seq_exec, RAJA::seq_exec >
+  >;
+
+// ----------------------------------------------------------------------------
+// OpenMP
+// ----------------------------------------------------------------------------
+#if defined(RAJA_ENABLE_OPENMP)
+
+using OpenMPKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::omp_for_exec>,
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::omp_parallel_for_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_exec, RAJA::seq_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_exec, RAJA::omp_parallel_for_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_static<8>, RAJA::seq_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_static<8>, RAJA::omp_parallel_for_exec >,
+
+    // Collapse Exec Pols
+    NestedLoopData<DEPTH_2_COLLAPSE, RAJA::omp_parallel_collapse_exec >,
+
+    // Depth 3 Exec Pols
+    NestedLoopData<DEPTH_3, RAJA::omp_parallel_for_exec, RAJA::loop_exec, RAJA::loop_exec >
+  >;
+
+#endif  // RAJA_ENABLE_OPENMP
+
+
+// ----------------------------------------------------------------------------
+// OpenMP Target
+// ----------------------------------------------------------------------------
+#if defined(RAJA_ENABLE_TARGET_OPENMP)
+
+using OpenMPTargetKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::omp_target_parallel_for_exec<16> >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::omp_target_parallel_for_exec<16> >,
+
+    // Collapse Exec Pols
+    NestedLoopData<DEPTH_2_COLLAPSE, RAJA::omp_target_parallel_collapse_exec >,
+
+    // Depth 3 Exec Pols
+    NestedLoopData<DEPTH_3, RAJA::seq_exec,  RAJA::omp_target_parallel_for_exec<16>, RAJA::seq_exec >
+  >;
+
+#endif  // RAJA_ENABLE_TARGET_OPENMP
+
+
+// ----------------------------------------------------------------------------
+// TBB
+// ----------------------------------------------------------------------------
+#if defined(RAJA_ENABLE_TBB)
+
+using TBBKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::tbb_for_exec >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::tbb_for_exec >,
+    NestedLoopData<DEPTH_2, RAJA::tbb_for_exec, RAJA::tbb_for_exec >,
```

Comment:
Using nested TBB threading policies here and below may be problematic similar to comment above for OpenMP.

Reply:
This should(tm) be ok.  It will create more tasking overhead, but the TBB constructs are meant to be nestable like this without issue, and they never create new parallelism explicitly, so this might actually be an ok test.

Reply:
Thanks again. I wasn't sure what was considered good usage for TBB along these lines.

Reply:
NP @rhornung67.  Truth be told it would rarely be good usage to do this, but if you had an outer loop with relatively low trip-count and an inner with a large one it might make sense to do it this way, more opportunity for the work-stealing to help.  It's not like with OpenMP where the best you could hope for is "not *much* worse" though, so may as well test it to make sure we don't have some breakage if someone wants to do it.